### PR TITLE
feat: Add full names in the description

### LINKED ISSUE #10194: [CAL-2189] change: "Dynamic Collective 60min Event" to "Firstname, Firstname & Firstname"
lets change this to include the peoples firstnames instead.

if its 2: "Firstname1 & Firstname2

if its more, it's "Firstname 1, Firstname2 & more"

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2189](https://linear.app/calcom/issue/CAL-2189/change-dynamic-collective-60min-event-to-firstname-firstname-and)</sub>

### PR DESCRIPTION
**Pull request by imrishabh18**
## What does this PR do?

Fixes #10194 
/claim #10194 

Screenshot 2023-07-19 at 3 18 36 AM

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Comment by imrishabh18** [comment]
@PeerRich Can you take a look at my approach to solve this please?

File: packages/features/bookings/Booker/components/EventMeta.tsx [hunk 1]
```diff
@@ -68,7 +70,11 @@ export const EventMeta = () => {
           <EventTitle className="my-2">{event?.title}</EventTitle>
           {event.description && (
             <EventMetaBlock contentClassName="mb-8 break-words max-w-full max-h-[180px] scroll-bar pr-4">
-              <div dangerouslySetInnerHTML={{ __html: markdownToSafeHTML(event.description) }} />
+              {defaultEvents.find((defaultEvent) => defaultEvent.description === event.description) ? (
```
Comment (jyellapu) [inline packages/features/bookings/Booker/components/EventMeta.tsx:73]: Please check , if you can use "isDynamic" flag present in the event to determine which path to take than relying on defaultEvent description.

File: packages/features/bookings/Booker/utils/layout.ts [hunk 2]
```diff
@@ -4,3 +4,19 @@ import { bookerLayoutOptions } from "@calcom/prisma/zod-utils";
...
+
+export const userNamesInDescription = (users) => {
+  const numberOfUsers = users.length;
+  if (numberOfUsers === 1) {
+    return `${users[0].name}`;
+  } else if (numberOfUsers === 2) {
+    return `${users[0].name} and ${users[1].name}`;
+  } else {
+    let listOfUsers = "";
+    for (let i = 0; i < numberOfUsers - 1; i++) {
```
Comment (jyellapu) [inline packages/features/bookings/Booker/utils/layout.ts:16]: Please check the ticket description, it's mentioned
"if its more, it's "Firstname 1, Firstname2 & more"

File: packages/features/bookings/Booker/utils/layout.ts [hunk 3]
```diff
@@ -4,3 +4,19 @@ import { bookerLayoutOptions } from "@calcom/prisma/zod-utils";
 export const validateLayout = (layout?: BookerLayouts | null) => {
   return bookerLayoutOptions.find((validLayout) => validLayout === layout);
 };
+
+export const userNamesInDescription = (users) => {
+  const numberOfUsers = users.length;
+  if (numberOfUsers === 1) {
```
Comment (jyellapu) [inline packages/features/bookings/Booker/utils/layout.ts:10]: Please handle numUsers == 0 case also.

**Review by jyellapu — CHANGES_REQUESTED** [review]
Please check the comments.

**Review by PeerRich — CHANGES_REQUESTED** [review]
the initial ticket was meant for only "dynamic booking links" only, not for all events

**Comment by PeerRich** [comment]
this is the correct implementation: https://github.com/calcom/cal.com/pull/10251 thank you for trying tho @imrishabh18

**Pull request closed without merging**